cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -4191,5 +4226,107 @@ TEST(TestArrowReadDeltaEncoding, DeltaBinaryPacked) {
 }
 #endif
 
+struct StatisticsTestParam {
+  std::shared_ptr<::arrow::Table> table;
+  int expected_null_count;
+  // This is the non-null count and not the num_values in the page headers.
+  int expected_value_count;
+  std::string expected_min;
+  std::string expected_max;
+};
+
+class ParameterizedStatisticsTest : public ::testing::TestWithParam<StatisticsTestParam> {
+};
+
+std::string GetManyEmptyLists() {
+  std::string many_empty_lists = "[";
+  for (int i = 0; i < 2000; ++i) {
+    many_empty_lists += "[],";
+  }
+  many_empty_lists += "[1,2,3,4,5,6,7,8,null]]";
+  return many_empty_lists;
+}
+
+TEST_P(ParameterizedStatisticsTest, NoNullCountWrittenForRepeatedFields) {
+  std::shared_ptr<::arrow::ResizableBuffer> serialized_data = AllocateBuffer();
```

Comment:
Would be nice to add a comment with a reference to the JIRA ticket.

Reply:
done.